pipelines/batch/src/main/java/com/google/fhir/analytics/JdbcResourceWriter.java

```diff
@@ -171,6 +206,9 @@ public void writeResource(Resource resource) throws SQLException, ViewApplicatio
           }
           ViewApplicator applicator = new ViewApplicator(vDef);
           RowList rowList = applicator.apply(resource);
+          // We should first delete old rows produced from the same resource in a previous run:
+          deleteOldViewRows(
+              jdbcDataSource, vDef.getName(), ViewApplicator.getIdString(resource.getIdElement()));
```

Comment:
nit: Can we do the `deletion` and `creation` of the records atomically to avoid partial failures, as well as the transaction is optimized because the commit is made at the end.